server/data_processors/DashboardDataProcessor/index.js

```diff
@@ -15,7 +15,9 @@ class DashboardDataProcessor {
   #dataFromAssessments = async () =>
     await Promise.all(
       this.#assessmentsFromConfig.map(async ({ collection, assessment }) => {
```

Comment:
I think now we don't need the #assessmentsFromConfig we can use the config directly and we don't need to append the collection property anymore. so the data result is all we would need now for the query. Perhaps a new list of just assessments from the config for a query that looks like. 

```javascript
const assessmentsFromConfig = config.map(({analysis} => analysis)

this.db.collection(collections.assessmentSubjectDayData)
          .find({ subject: req.params.subject, assessment: { $in: assessmentsFromConfigs }).toArray()
```
```suggestion
  #dataFromAssessments = async () =>
    await Promise.all(
      this.#assessmentsFromConfig.map(async ({ collection, assessment }) => {
```

Reply:
I think you're right but there's a lot that can be done to improve performance and legibility here that I don't want to block being able to update the underlying collection. My preference would be to handle this as part of a future performance improvement PR.

Reply:
that makes sense, but I do feel like we aren't missing the subject in the query

Reply:
e42ee3084c5d27c1c4ba2f3b3590560d8f5fa3bf